internal/app/go-filecoin/storage_miner_connector/connector.go

```diff
@@ -303,64 +305,67 @@ func (m *StorageMinerNodeConnector) GetSealTicket(ctx context.Context) (storage.
 // responsible for choosing an interval-value, which is a quantity of blocks to
 // wait (after the block in which the pre-commit message is mined) before
 // computing and sampling a seed.
-func (m *StorageMinerNodeConnector) GetSealSeed(ctx context.Context, preCommitMsg cid.Cid, interval uint64) (seed <-chan storage.SealSeed, err <-chan error, invalidated <-chan struct{}, done <-chan struct{}) {
-	sc := make(chan storage.SealSeed)
-	hc := make(chan block.TipSetKey)
-	ec := make(chan error)
-	ic := make(chan struct{})
-	dc := make(chan struct{})
+func (m *StorageMinerNodeConnector) GetSealSeed(ctx context.Context, preCommitMsg cid.Cid, interval uint64) (<-chan storage.SealSeed, <-chan error, <-chan struct{}, <-chan struct{}) {
+	seedCh := make(chan storage.SealSeed)
+	errCh := make(chan error)
+	invalidCh := make(chan struct{})
+	doneCh := make(chan struct{})
 
 	go func() {
 		h, exitCode, err := m.waitForMessageHeight(ctx, preCommitMsg)
 		if err != nil {
-			ec <- err
+			errCh <- err
 			return
 		}
 
 		if exitCode != 0 {
-			ec <- xerrors.Errorf("non-zero exit code for pre-commit message %d", exitCode)
+			errCh <- xerrors.Errorf("non-zero exit code for pre-commit message %d", exitCode)
 			return
 		}
 
-		m.chainHeightScheduler.AddListener(h + interval)
-	}()
+		listener := m.chainHeightScheduler.AddListener(h + interval)
 
-	// translate tipset key to seal seed handler
-	go func() {
+		// translate tipset key to seal seed handler
 		for {
 			select {
-			case key := <-hc:
+			case key := <-listener.HitCh:
 				ts, err := m.chainState.GetTipSet(key)
 				if err != nil {
-					ec <- err
-					break
+					listener.ErrCh <- err
```

Comment:
hmm do we deadlock here as the same goroutine sends and receives on this channel?

Reply:
Yep, this should be errCh, not listener.ErrCh